### ISSUE TITLE
Package Manager Console Can Freeze Visual Studio During A Build And The User Cannot Cancel The Build

### ISSUE THREAD

**Issue by vsfeedback** [aspnet/EntityFramework6#1428]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/794815/package-manager-console-can-freeze-visual-studio-d.html)._

---
<p><strong>Problem:</strong><br>
Starting with Visual Studio 2019, I have noticed that the package manager console can take a very long time to build a project and if it encounters an error during the build, it takes even longer for it to stop what it is doing before it reports there was an error. During the time it is building, Visual Studio becomes entirely frozen and you cannot click the button to stop the build and Visual Studio also does not respond to CTRL+BREAK. The only way to stop it is to go into the task manager and kill any processes of MSBuild.EXE.</p>
<p><strong>One simple way to do this (I imagine there are a lot of other ways) is to use these steps:</strong></p>
<ol>
<li>Create a class library on .NET Core 3.0 for Entity Framework Core and put in a model and DB Context into it.</li>
<li>Create a separate console app target .NET Core 3.0 that has the necessary dependencies for creating Entity Framework migrations and have it reference the class library.</li>
<li>Create another console app that does not have any dependency on the Entity Framework class library.</li>
<li>Make the console app from step 2 your startup project</li>
<li>Open the package manager console</li>
<li>In the drop down at the top of the package manager console, select the WRONG project for your default project. So in this case, select the console app made in step 4.</li>
<li>In the package manager console, enter “Add-Migration InitialCreate -Context [DbContextNameFromStep1] -OutPutDir [PathToMigrationsFolderInYourClassLibrary]”</li>
<li>Hit enter to run the migration command.</li>
</ol>
<p><strong>Result:</strong></p>
<p>Visual studio will take forever to “build” and then eventually it will fail because the default project specified doesn’t have the correct dependencies to create the migration. While it is taking forever, VS does not respond to mouse clicks and you cannot stop the build with CTRL+BREAK.</p>
<p>As previously mentioned, the only way to stop the build is to kill the MsBuild.exe processes running in the task manager. Visual Studio will then stop building and say the build failed.</p>
<p>This is just one specific repro, as I said, I have noticed other instances where while the package manager is running a command, Visual Studio becomes unresponsive and you cannot do anything in it until the package manager is finished.</p>
<p>If you need further details and/or if this issue has already been reported or anything else, let me know.</p>

---
#### Fiona Niu[MSFT] on 10/28/2019, 03:07 AM: 

<p>Thank you for taking the time to log this issue!<br>
I’ve tried to reproduce and investigate using the description, and attachments already provided. Unfortunately those aren’t enough and more information is needed in order to investigate it further.<br>
The easiest way to provide all the information is to use the Visual Studio Feedback Tool. This will ensure that we collect the needed information for you without worrying about what to provide (recording, dump file or ETL trace).<br>
Since this issue is now marked as Need More Info, that workflow is enabled in the Feedback Tool:</p>
<pre><code>• Open Visual Studio Feedback tool.
• Click the banner letting you know that you have problems requesting your attention.
• Click this problem from the list
• Click &quot;View their request and respond&quot; from the problem details banner
• Add a comment, in the Attachments/Record: click Start Recording
• When the Steps Recorder tool appears, perform the steps that reproduce the problem.
• When you're done, choose the Stop Record button.
• Wait a few minutes for Visual Studio to collect and package the information that you recorded.
• Submit. You will be able to see the comment on Developer Community. For security reasons, your files come directly to us and don't appear on Developer Community.
</code></pre>
<p>For the full instructions, please see: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio-2017?view=vs-2017#when-further-information-is-needed-need-more-info">https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio-2017?view=vs-2017#when-further-information-is-needed-need-more-info</a> . For information about what data is collected, see <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/developer-community-privacy?view=vs-2017#data-we-collect">https://docs.microsoft.com/en-us/visualstudio/ide/developer-community-privacy?view=vs-2017#data-we-collect</a><br>
We look forward to hearing from you!</p>

#### Stephen White on 10/28/2019, 07:09 AM: 

<p>Thanks Fiona, I will try out what you are asking later today. </p><p><br>Before I attempt a reproduction, do you know if it will cause issues with the recorder if I end the MS build processes to get the Package Manager to yield control back to the UI or should I wait until the package manager fails to build on its own? </p>

#### Fiona Niu[MSFT] on 10/29/2019, 03:45 AM: 

<p>@<a target='_blank' href="https://developercommunity.visualstudio.com/users/106380/StephenWhite.html">Stephen White</a> Thanks for your reply. For VS hang/freeze issue, please start the VS feedback tool before VS becomes not responding, or start a new Visual Studio instance.</p>

---
### Original Solutions
(no solutions)

**Comment by ajcvickers** [comment]
Please post a small, runnable solution that demonstrates the behavior you are seeing so that we can investigate.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by tgp1994** [comment]
I'd like to have this issue reopened, I just ran into it myself. I ran a `Update-Package -reinstall` command in the **Package Manager Console**, then went to do other things while it was running. When I came back to Visual Studio, I thought it had finished - even though the command was waiting for user input. Being forgetful, I tried building my project, but like the original reporter states, Visual Studio will just sit in a _building_ status with no output whatsoever or any indication of why it's hung. I could still cancel the build, but it wasn't until I opened the Package Manager Console that I realized it was waiting for my input. I finished the command, and building could resume after.

To reproduce this, in a project with any NuGet packages, run the command `Update-Package -reinstall` in a scenario where it needs to gather user input. For example, a package file had been modified and needed my input on what to do. Let the input wait, and attempt to build the project, and Visual Studio will sit in a Building state with no indication of what is going on. Ideally it would be great is Visual Studio could alert us that a command is blocking the Build process from proceeding.
